packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -353,3 +358,27 @@ export class ChainService implements ChainServiceInterface {
     return result;
   }
 }
+
+async function connectContract(
+  address: Address,
+  {
+    deployedBytecode,
+    abi,
+    contractName,
+  }: {contractName: string; abi: ethers.ContractInterface; deployedBytecode: string},
+  provider: providers.JsonRpcProvider
+): Promise<Contract> {
+  const bytecode = await provider.getCode(address);
+
+  // TODO: Throw errors once we can be certain the bytecode being referenced is correct;
+  //       right now it may be incorrect since our ContractArtifacts refer to local deployments
+  //       and not deployed bytecode on various other chains e.g., rinkeby, mainnet, etc
+
+  if (BN.eq(bytecode, 0))
```

Comment:
If I understand correctly, bytecode is not a number.

Reply:
It's a `Bytes` which is a `BigNumberish`

Reply:
Might be easier to do a direct string comparison instead like `== "0x00"`.

Reply:
Perhaps it's pedantic, but it would be nice to use `BigNumber` and `BigNumberish` for variables/constants that represent numbers.